.github/workflows/downstream_inclusion.yml

```diff
@@ -0,0 +1,148 @@
+      TEST_PROJECT_DIR: submodule_test_project
+    steps:
+
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Setup uv
+      uses: yezz123/setup-uv@v4
+
+    - name: Checkout nias-cpp
+      uses: actions/checkout@v4
+      with:
+        path: nias-cpp
+
+    - name: Create test project
+      run: |
+        mkdir "${TEST_PROJECT_DIR}"
+        cp nias-cpp/.test_projects/main.cpp "${TEST_PROJECT_DIR}/"
+        cp nias-cpp/tests/test_vector.h "${TEST_PROJECT_DIR}/"
+        cp nias-cpp/.test_projects/CMakeLists.txt.Submodule "${TEST_PROJECT_DIR}/CMakeLists.txt"
+
+    - name: Checkout nias_cpp as submodule
+      uses: actions/checkout@v4
+      with:
+        path: ${{ env.TEST_PROJECT_DIR }}/nias-cpp
+
+    - name: Configure and build test project
+      run: |
+        cd "${TEST_PROJECT_DIR}"
+        mkdir build
+        cmake -B build -DPython_EXECUTABLE="$(which python3.12)" -DCMAKE_BUILD_TYPE=Debug
+        cmake --build build
+
+  inclusion_via_fetchcontent:
+    strategy:
+      fail-fast: false
+    name: using FetchContent
+    runs-on: ubuntu-24.04
+    defaults:
+      run:
+        shell: bash -elo pipefail {0}
+    timeout-minutes: 30
+    env:
+      TEST_PROJECT_DIR: fetchcontent_test_project
+    steps:
+
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Setup uv
+      uses: yezz123/setup-uv@v4
+
+    - name: Checkout nias-cpp
+      uses: actions/checkout@v4
+      with:
+        path: nias-cpp
+
+    - name: Create test project
+      run: |
+        mkdir "${TEST_PROJECT_DIR}"
+        cp nias-cpp/.test_projects/main.cpp "${TEST_PROJECT_DIR}/"
+        cp nias-cpp/tests/test_vector.h "${TEST_PROJECT_DIR}/"
+        cp nias-cpp/.test_projects/CMakeLists.txt.FetchContent "${TEST_PROJECT_DIR}/CMakeLists.txt"
+        sed -i 's/<GIT_TAG_PLACEHOLDER>/${{ github.event.pull_request.head.sha }}/' "${TEST_PROJECT_DIR}/CMakeLists.txt"
+
+    - name: Configure and build test project
+      run: |
+        cd "${TEST_PROJECT_DIR}"
+        mkdir build
+        cmake -B build -DPython_EXECUTABLE="$(which python3.12)" -DCMAKE_BUILD_TYPE=Debug
+        cmake --build build
+
+  use_from_virtualenv:
+    strategy:
+      fail-fast: false
+    name: from virtualenv
+    runs-on: ubuntu-24.04
+    defaults:
+      run:
+        shell: bash -elo pipefail {0}
+    timeout-minutes: 30
+    env:
+      TEST_PROJECT_DIR: virtualenv_test_project
+      PYTHON_VERSION: 3.12
+    steps:
+
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+
+    - name: Setup uv
+      uses: yezz123/setup-uv@v4
+
+    - name: Checkout nias-cpp
+      uses: actions/checkout@v4
+      with:
+        path: nias-cpp
+
+    - name: Setup virtualenv
+      run: |
+        uv venv --python=${{ env.PYTHON_VERSION }}
+        uv pip install ./nias-cpp
```

Comment:
Makes the setup-python step not strictly required (uv would download if no matching installed)

Reply:
Right, but I still kept it to be able to use the python-path output.